### DISCUSSION TITLE
A lot of package do not hit the binary cache

### DISCUSSION THREAD

**Discussion by benneti** [nrdxp/nixdev#363] [Q&A]
What do you think is the reason that I need to compile a lot of packages (even gnome and firefox-wayland) that I should get from the 21.05 cache locally?
Could it be related to the ANTI CORRUPTION LAYER?

**Comment by benneti** [comment]
After seeing https://github.com/divnix/digga/commit/7115e02b64f86a78e47477c16a121629269f3988 @blaggacao might be interested.
I have digga locked to the above commit and still these mass rebuilds occured, might be related

**Comment by nrdxp** [comment]
we probably need to setup an auto update to the nixos input in CI because it's a bit old and if you've updated it on your own system then you'll miss the cache pretty much all the time

**Reply by blaggacao** [reply]
Good point! In the absence of Comtent Adressed Stores, that might be our only option. - assuming rebuilds often occur due to negligable input changes.

**Reply by benneti** [reply]
Yeah an auto update for nixos input would be appreciated!
Maybe https://github.com/nix-community/emacs-overlay could give some inspiration, a few month ago they did something that enabled hitting the binary cache basically always (instead of basically never before), but I am not sure what it was as I do not usually use CI.

**Reply by benneti** [reply]
See also here https://github.com/nix-community/emacs-overlay/issues/122

**Comment by blaggacao** [comment]
W.r.t. anti-corruption layer: we might be able to remove it and simoly require people to run a recent `nix` version?

That would solve a lot of headache.

Though I don't think it's related.

**Reply by benneti** [reply]
I think this is a reasonable requirement as long as it is communicated clearly.
As running flakes already kinda implies that one would like to use a recent `nix` version.

**Comment by benneti** [comment]
It seems to me that the cache is hit normally again!

**Reply by nrdxp** [reply]
some of the locks were recently update thanks to @GTrunSec :bow:
